src/operator/nn/mkldnn/mkldnn_rnn.cc

```diff
@@ -676,7 +676,8 @@ void MKLDNNRnnOp::Init(const OpContext &ctx,
       "Layer vector's size has a different value than the number of fusion.";
   if (dst_.size() < num_fusion - 1) {
     int data_dtype = outputs[rnn_enum::kOut].dtype();
-    for (auto fwd = fwd_inf_vec_.begin(); fwd < fwd_inf_vec_.end() - 1; ++fwd)
+    for (std::vector<MKLDNNRnnForward>::const_iterator fwd = fwd_inf_vec_.begin();
+        fwd < fwd_inf_vec_.end() - 1; ++fwd)
```

Comment:
strange code style. Use `fwd != fwd_inf_vec_.end() - 1` instead. For curious, why fwd can't be fwd_inf_vec_.end() - 1? Shouldn't we remove the last element from fwd_inf_vec_?

Reply:
Here we need `fwd_inf_vec_.size() - 1` spaces for the intermediate results of the multiple fused layers. And for the result of the last fused layer, `outputs[rnn_enum::kOut]` could provide the space. Hence, `forward_inf_vec_.back()` is excluded when allocates the spaces for intermediate results.

Reply:
Please make your reply as a code comment here. Then other developer will know the reason behind by reading it.

Reply:
Thanks for reminding me. I will add the comment.